matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/di/WorkManagerProvider.kt

```diff
@@ -60,6 +74,27 @@ internal class WorkManagerProvider @Inject constructor(
         }
     }
 
+    private fun checkIfWorkerFactoryIsSetup() {
+        sessionScope.launch(coroutineDispatchers.main) {
+            val checkWorkerRequest = OneTimeWorkRequestBuilder<MatrixWorkerFactory.CheckFactoryWorker>().build()
+            workManager.enqueue(checkWorkerRequest)
+            val checkWorkerLiveState = workManager.getWorkInfoByIdLiveData(checkWorkerRequest.id)
+            val observer = object : Observer<WorkInfo> {
+                override fun onChanged(workInfo: WorkInfo) {
+                    if (workInfo.state.isFinished) {
+                        checkWorkerLiveState.removeObserver(this)
+                        if (workInfo.state == WorkInfo.State.FAILED) {
+                            throw RuntimeException("MatrixWorkerFactory is not being set on your worker configuration.\n" +
+                                    "Makes sure to add it to a DelegatingWorkerFactory if you have your own factory or use it directly.\n" +
+                                    "You can grab the instance through the Matrix class.")
```

Comment:
The message is maybe a bit hard to understand (to me at least) :/. I cannot propose better though, but maybe add a link to the Element code where it's done?